### PR TITLE
Add security events

### PR DESCRIPTION
**Pull request by skeet70**

### CODE REVIEW

**Review by clintfred — APPROVED** [review]
LGTM

File: src/security-events/EventMetadata.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import {clearUndefinedProperties} from "../Util";
+
+export interface EventMetadataJson {
+    tenantId: string;
+    timestampMillis: number;
+    iclFields: {
+        requestId?: string;
+        event?: string;
+        sourceIp: string | undefined;
+        objectId: string | undefined;
```
Comment (BobWall23) [inline src/security-events/EventMetadata.ts:10]: How is a `sourceIp: string | undefined` different from `sourceIp?: string`?

File: src/security-events/EventMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        requestingId: string;
```
Comment (BobWall23) [inline src/security-events/EventMetadata.ts:11]: Might be helpful to maintain consistency and call this `requestingUserOrServiceId`

File: src/security-events/EventMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            timestampMillis: this.timestampMillis,
+            iclFields: {
+                sourceIp: this.sourceIp,
+                objectId: this.objectId,
+                requestingId: this.requestingUserOrServiceId,
+                dataLabel: this.dataLabel,
+            },
+            customFields: this.otherData,
+        };
+        if (this.requestId) {
```
Comment (BobWall23) [inline src/security-events/EventMetadata.ts:92]: Can't this be handled the same way as the other optional fields - just assign it, then let `clearUndefinedProperties` get rid of it if it is undefined?

**Review by BobWall23 — APPROVED** [review]

File: src/security-events/EventMetadata.ts (hunk 1 above)
Comment (skeet70) [reply src/security-events/EventMetadata.ts:10]: It's not. They're the same.

File: src/security-events/EventMetadata.ts (hunk 2 above)
Comment (skeet70) [reply src/security-events/EventMetadata.ts:11]: Good catch.

Comment (skeet70) [reply src/security-events/EventMetadata.ts:11]: Oh sorry, thought this was on the constructor. No, that would be inconsistent. The public interface is the only place it's called that, and it's the same as the Java side. It'd require many more changes, and not for any benefit to us.

File: src/security-events/EventMetadata.ts (hunk 3 above)
Comment (skeet70) [reply src/security-events/EventMetadata.ts:92]: Yeah, I'll change that, it was written before the clear was. I should also change the throws, the Java code only throws if it doesn't have a tenantId.

**Review by giarc3 — APPROVED** [review]

**Review by BobWall23 — APPROVED** [review]

**Pull request merged**